pkg/operator/status.go

```diff
@@ -0,0 +1,151 @@
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions,
+		operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeProgressing,
+			Status:  operatorv1.ConditionTrue,
+			Message: message,
+		})
+
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeFailing,
+		Status: operatorv1.ConditionFalse,
+	})
+
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions,
+		operatorv1.OperatorCondition{
+			Type:   operatorv1.OperatorStatusTypeAvailable,
+			Status: operatorv1.ConditionFalse,
+		})
+}
+
+func (c *serviceCAOperator) setAvailableStatus(operatorConfig *operatorv1.ServiceCA) {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeAvailable,
+		Status: operatorv1.ConditionTrue,
+	})
+
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeProgressing,
+		Status: operatorv1.ConditionFalse,
+	})
+
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeFailing,
+		Status: operatorv1.ConditionFalse,
+	})
+}
+
+func isDeploymentStatusAvailable(deploy *appsv1.Deployment) bool {
+	return deploy.Status.AvailableReplicas > 0
+}
+
+// isDeploymentStatusAvailableAndUpdated returns true when at least one
+// replica instance exists and all replica instances are current,
+// there are no replica instances remaining from the previous deployment.
+// There may still be additional replica instances being created.
+func isDeploymentStatusAvailableAndUpdated(deploy *appsv1.Deployment) bool {
+	return deploy.Status.AvailableReplicas > 0 &&
+		deploy.Status.ObservedGeneration >= deploy.Generation &&
+		deploy.Status.UpdatedReplicas == deploy.Status.Replicas
+}
+
+func isDeploymentStatusComplete(deploy *appsv1.Deployment) bool {
+	replicas := int32(1)
+	if deploy.Spec.Replicas != nil {
+		replicas = *(deploy.Spec.Replicas)
+	}
+	return deploy.Status.UpdatedReplicas == replicas &&
+		deploy.Status.Replicas == replicas &&
+		deploy.Status.AvailableReplicas == replicas &&
+		deploy.Status.ObservedGeneration >= deploy.Generation
+}
+
+func (c *serviceCAOperator) syncStatus(operatorConfigCopy *operatorv1.ServiceCA, deployments []string) (bool, error) {
+	ready := 0
+	existingDeploymentsAndReplicas := 0
+	var statusErr error
+	for _, dep := range deployments {
+		existing, err := c.appsv1Client.Deployments(operatorclient.TargetNamespace).Get(dep, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			glog.Infof("Deployment %s does not exist", dep)
+			statusMsg := fmt.Sprintf("Deployment %s does not exist", dep)
+			c.setFailingStatus(operatorConfigCopy, "Error getting service-ca-operator managed deployments", statusMsg)
+			return false, statusErr
+		} else if existing.DeletionTimestamp != nil {
```

Comment:
Each of these if's end with return, no need to do `else if`

Reply:
yes, thanks : ).  I've kept them all as if statements (no else), instead of switch.